### DISCUSSION TITLE
Google Coral M.2

### DISCUSSION THREAD

**Discussion by veno77** [blakeblackshear/frigate#1737] [Ask A Question] [answered]
Hello,

I have Asus PN40 Mini-PC with Intel N4020 CPU running Debian with HA Supervised and Frigate in Docker Container. 
I want to add Coral Edge TPU for monitoring 4 x 1080P cameras. The Asus PN40 have M.2 NVMe 2280 slot and 2230 slot for WIFI/BT card which I don't need. Can I put in Coral M.2 B+M 2280 or M.2 A+E 2230 in these slots? 

Thanks!

**Comment by veno77** [comment, marked as answer]
Just to answer my question. I ordered Coral M.2 B+M 2280 card and it is working flawlessly in ASUS PN40.
I think the M.2 A+E 2230 card will also work in the slot for WIFI/Bluetooth card. If I need one more probably I will order the later one.
The only issue I am facing is that due to the small form factor and passive cooling, the temperature raised from 55-60 usually to over 70*C . The cpu load with 4 Full HD cameras, with detect resolution at 1280x720 and 5fps is ~30% (this includes the supervised HASS with numerous integrations and several other containers outside HASS like MariaDB and Frigate). Pretty good for dual core low power (6W) celeron.

I am planning drilling holes in the top cover to ease the airflow, if this did not help much I can put slow rpm 120mm fan on top of it.

**Reply by TurboTronix** [reply]
Are you using it with docker within win10?

**Reply by veno77** [reply]
No, I use it with docker in Debian Linux.

**Reply by R5fan** [reply]
> I think the M.2 A+E 2230 card will also work in the slot for WIFI/Bluetooth card.

Indeed it does. Single TPU only but thats really plenty.

> If I need one more 

I doubt it :). With 6 camera's I have not managed to get inference speed over 30ms with single core TPU. My CPU may start struggling with decoding and motion detection, but that coral is basically asleep.  I dont think for frigate it even makes sense to get a dual core TPU, other than just for the form factor/slot.

**Reply by blakeblackshear** [reply]
To clarify, the inference speeds should not change with the number of cameras. You want to monitor your detection fps to see if you are hitting the limits of the coral. See last paragraph here: https://github.com/blakeblackshear/frigate/blob/release-0.9.0/docs/docs/hardware.md#google-coral-tpu

**Reply by tortho** [reply]
> > I think the M.2 A+E 2230 card will also work in the slot for WIFI/Bluetooth card.
> 
> Indeed it does. Single TPU only but thats really plenty.
> 
M.2-2230-D3-E module do not work for me on HomeAssistantOS 7.0RC1. Frigate detects the module but get errors and unusable. Do you run as supervised or other installation method?

**Comment by AaronAutomated** [comment]
Did you ever test the A+E chip in the WiFi / BT M.2 E Key lane? I bought the Dual TPU and placed into my M.2 E Key for WiFi / BT but the device isn't being detected at all..

**Comment by tortho** [comment]
Im using it in the M.2 E key for wifi/bt and working successfully under the latest HAOS
